Classes/Format/Mods.php

```diff
@@ -78,56 +119,216 @@ public function extractMetadata(\SimpleXMLElement $xml, array &$metadata)
-                    $metadata['year_sorting'][0] = intval($year_sorting);
+                    $this->metadata['year_sorting'][0] = intval($year_sorting);
                 }
             }
         }
     }
+
+    /**
+     * Get "description" stored in recordInfoNote.
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getDescription() {
+        $this->getSingleMetadata('description', './mods:recordInfo/mods:recordInfoNote/text()');
+    }
+
+    /**
+     * Get "identifier" - example: hotels (built public accommodations), AAT ID: 300007166.
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getIdentifier() {
+        $this->getSingleMetadata('identifier', './mods:identifier/text()');
+    }
+
+    /**
+     * Get "license" - license on which object is published.
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getLicense() {
+        $this->getSingleMetadata('license', './mods:accessCondition/text()');
+    }
+
+    /**
+     * Get names of the original object:
+     *   - main name
+     *   - alternative names
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getObjectNames() {
+        $this->getSingleMetadata('object_name', './mods:relatedItem/mods:titleInfo[not(@displayLabel="alternative")]/mods:title/text()');
+        $this->getSingleMetadata('object_alternative_names', './mods:relatedItem/mods:titleInfo[@displayLabel="alternative"]/mods:title/text()');
+    }
+
+    /**
+     * Get location information about the original object:
+     *  - city (object_location)
+     *  - geonames
+     *  - wikidata
+     *  - wikipedia
+     *
+     * @access private
+     *
+     * @return void
+     */
+    private function getObjectLocationMetadata() {
+        $this->getSingleMetadata('object_location', './mods:relatedItem/mods:location/mods:physicalLocation/text()');
+        $this->getSingleMetadata('geonames', './mods:relatedItem/mods:location/mods:url[@displayLabel="geonames"]/text()');
+        $this->getSingleMetadata('wikidata', './mods:relatedItem/mods:location/mods:url[@displayLabel="wikidata"]/text()');
+        $this->getSingleMetadata('wikipedia', './mods:relatedItem/mods:location/mods:url[@displayLabel="wikipedia"]/text()');
+    }
+
+    /**
+     * Save to the metadata array the last found matching metadata.
+     *
+     * @access private
+     *
+     * @param string $metadataIndex: The index in the array by which metadata is going to be accessible
+     * @param string $xpath: The xpath for searching metadata in MODS
+     *
+     * @return void
+     */
+    private function getSingleMetadata($metadataIndex, $xpath) {
+        $results = $this->xml->xpath($xpath);
+        if (!empty($results)) {
+            foreach ($results as $result) {
+                $this->metadata[$metadataIndex][0] = (string) $result;
+            }
+        }
+    }
```

Comment:
See above: this method isn't needed if all single metadata mappings are moved to the default configuration.